feat(cli): show table for getorders

erkarl opened the pull request:
Related: https://github.com/ExchangeUnion/xud/issues/558

This PR modifies the getorders cli to return results in a table:
![screen shot 2018-10-24 at 16 23 27](https://user-images.githubusercontent.com/1056569/47433526-4a40b680-d7a9-11e8-9da3-0b97f0199f40.png)

With the current implementation all trading pairs are displayed in the same table and can be distinguished by "Pair" column. Should we keep it this way or display a separate table for each pair?

For drawing the table I chose [cli-table3](https://github.com/cli-table/cli-table3) dependency as it's based on cli-table and cli-table2 which are no longer maintained. Additionally cli-table3 has TypeScript support.

I'm thinking we should add additional cli `--raw` option for users who want to receive raw output (perhaps for debugging purposes?).

Commit:
chore: add cli-table3 dependency

Commit:
feat(cli): extend cli callback with optional formatting cb

Changes callback to return a function instead so that we can inject an
optional formatting callback.

Commit:
feat(cli): show table as getorders output

sangaman commented:
Awesome! I pulled down the branch and played with it and it's looking nice. A couple things from a first pass:

- The `Time` column gets cut off for me, ending with `7:26:1…` instead of the full time. I'm thinking we maybe don't need this column tho, as I don't see it as important information just for seeing what's out there. I'm not used to seeing it on most orderbooks on actual exchanges (or if it's there, I don't notice). For me, it'd be nice to have but if it presents issues or takes up too much space we can drop it.

- Rather than have a column for buy/sell and a column for pair, I think it would be nice to have separate tables for each trading pair that is returned and to have all buys on the left side of the tables, and all sells on the right side of the table. Each side would have its own price, quantity, and own order column. 

- I think it would be nice to highlight our own orders a certain color, making it easy to see which orders are "out there" and which are ours. It could even serve as a replacement for the own order column if we don't have space, but then we'd need to make the meaning of the highlight clear to the user.

I really like the idea of a `--raw` option to just output the JSON as we do currently, that can be a global arg for all commands defaulting to `false`.

erkarl commented:
Without commiting code I'm posting another version for review. Based on the feedback I:
* removed the date column as it was taking up too much space
* removed the ID column since it wasn't really useful in terms of getting an overview of the orderbook
* moved buys to left side of the table & sells to the right
* removed own order column and marked own orders as cyan (I'm not sure if this is the best approach as it might not be apparent for everyone)

![screen shot 2018-11-02 at 12 39 14](https://user-images.githubusercontent.com/1056569/47910793-93c18d80-de9c-11e8-8a4d-374db7d2e4c0.png)

sangaman commented:
It feels like there'd be room for the Own Orders column maybe? Then the highlighting will also make sense. I think it looks great otherwise.

erkarl commented:
I think we can fit in "Own Order" column - the reason I didn't add it was because it felt weird having it "No" for 99% of the cases.

I'll add it for now. We can always remove it in the future when the network's orderbook grows.

Commit:
feat(cli): add global --json option/arg

erkarl commented:
Updated with code + added "Own Order" column.

sangaman commented on bin/xucli:
```diff
@@ -27,6 +27,12 @@ require('yargs')
       describe: 'Path to the TLS certificate of xud',
       type: 'string',
     },
+    raw: {
```
For this option, how about calling it `json` with a `-j` flag and description `Display output in json format`. My thinking is that the "raw" responses are serialized data and that the json output is actually formatted. This makes it clear that the output will be in JSON.

erkarl replied:
Calling it `json` instead of `raw` feels way better.

sangaman reviewed (commented):
Looks great, just a couple of things:

I see that you've colored the "Buy" side red and the "Sell" side green. I'm used to seeing these colors in reverse. Buys are green, sells are red.

Also, about having to print "No" over and over, you could maybe just put an "X" in the column for own orders and leave it empty for peer orders. Or you could only write "Yes" for own orders if "X" is confusing. Just an idea, I'd be fine with how it is currently.

erkarl commented:
@sangaman updated. Let me know if you have reviewed and agree with the changes so I can squash them.

sangaman reviewed (approved):
Awesome.

One idea I had, but not for this PR, would be to have a node's alias or node pubkey in the table. I'll open a separate issue to discuss it, but I think this will be a big improvement.
